config/rbac/hive_admin_role_binding.yaml

```diff
@@ -1,3 +1,6 @@
+# NOTE: This binding uses the openshift apigroup as it is the only way to link
```

Comment:
Would it be possible to use the `rbac.authorization.k8s.io` version in most cases and only the `authorization.openshift.io` version when running on OpenShift 3.11? The `rbac.authorization.k8s.io` version supports group. See https://github.com/openshift/api/blob/12f2193c359f84d4272827843d3efc2362d62344/authorization/v1/types.go#L397-L402.

Or is an OpenShift group different than a normal k8s group?

Reply:
I have no idea what the k8s group is, I can't find any such type in kind or openshift, I think it might be a separate concept and the two are very much different.

Reply:
See https://kubernetes.io/docs/reference/access-authn-authz/authentication/

As far as I know, there are not resources that define groups. There are many parts of the system that determine which groups a user belongs to.

Reply:
But the concept of a group should be the same regardless of which type of kube you are using.

Reply:
I don't want to risk cutting off our access in stg and prod, and I don't really want to invest the time in testing and debugging how this behaves if we make the switch. Propose we take this change and solve the kind problem for now, and invest in using just the rbac type in future if needed. These two roles are truly not needed in hive dev clusters.

Reply:
That's fine.

It would not affect stage and prod. They would both continue to use the same that they are using now because they are running on an older version of kubernetes that does not support a group kind for the rbac role binding.

My overarching concern is trying to push back on keeping SRE-specific functionality in the Hive repo. If these roles are not something that is needed to run Hive, then they should not be part of the Hive repo. If they are only needed to support some operation that SRE is doing, then SRE should be responsible for maintaining and applying those role bindings.

Reply:
I agree we should keep SRE specifics out and this one is borderline, we considered that at the time but reasoned that this was generically useful enough to consider fair game for including in the main repo.